### PR TITLE
fsync interval and filter.max_size settings added

### PR DESCRIPTION
**Pull request by pustovit**

### CODE REVIEW

File: docs/configuration.txt [hunk 1]
```diff
@@ -115,6 +115,29 @@ Translog
...
+
+**index.gateway.local.sync**
+  | *Default:*    ``5``
+  | *Runtime:*   ``yes``
+
+  Sets the period of translog fsyncing to the disk
+
+  .. note::
+
+  Keep in mind that data which was not ``fsync``ed to the disk can be lost in case of a power failure.
```
Comment (chaudum) [inline docs/configuration.txt]: please add more detailed information about this setting according to https://www.elastic.co/guide/en/elasticsearch/guide/master/translog.html as we discussed in the daily

File: docs/configuration.txt [hunk 2]
```diff
@@ -91,18 +91,33 @@ Blocks
...
 --------
 
+.. note:
+
+  The translog provides a persistent log of all operations that have not been transferred (flushed) to disk.
+  Whenever a record is inserted into a table (or updated) that change is appended both to an in-memory buffer **and** the translog.
+  When refreshing a table the changes from the in-memory buffer are written to a new commit point (which is then searchable)
+  and the buffer is cleared, unlike the translog. When the translog reaches a certain size (see :ref:`translog.flush_threshold_size`),
+  or holds a certain amount of operations (see :ref:`translog.flush_threshold_ops`), or after a certain interval
+  (see :ref:`translog.flush_threshold_period`) the translog is `fsync`ed, flushed to disk, and cleared.
```
Comment (seut) [inline docs/configuration.txt]: it's not cleared

File: docs/configuration.txt [hunk 3]
```diff
@@ -115,6 +130,41 @@ Translog
...
+**index.cache.filter.max_size**
+  | *Default:*    ``5``
+  | *Runtime:*   ``yes``
+
+The maximum size of filter cache (per segment in shard). Set to -1 to disable.
+
+Gateway
+-------
+
+**index.gateway.local.sync**
```
Comment (seut) [inline docs/configuration.txt]: this setting occurs twice

File: sql/src/main/java/io/crate/metadata/information/InformationTableInfo.java [hunk 4]
```diff
@@ -82,6 +82,10 @@
                 ImmutableList.of("warmer"));
         public static final ColumnIdent TABLE_SETTINGS_WARMER_ENABLED = new ColumnIdent("settings",
                 ImmutableList.of("warmer", "enabled"));
+        public static final ColumnIdent TABLE_SETTINGS_GATEWAY_LOCAL = new ColumnIdent("settings",
+                ImmutableList.of("gateway_local"));
```
Comment (seut) [inline sql/src/main/java/io/crate/metadata/information/InformationTableInfo.java]: doesn't it have to be `ImmutableList.of("gateway", "local")`?

File: docs/configuration.txt [hunk 5]
```diff
@@ -115,6 +130,41 @@ Translog
 
   Disable/enable flushing.
 
+Gateway
+-------
+
+**index.gateway.local.sync**
+  | *Default:*    ``5``
```
Comment (seut) [inline docs/configuration.txt]: change to '5s'

File: sql/src/test/java/io/crate/analyze/CreateAlterTableStatementAnalyzerTest.java [hunk 6]
```diff
@@ -761,6 +761,16 @@ public void testRecoveryShardsWithInteger() throws Exception {
     }
 
     @Test
+    public void testGatewayLocalSyncPeriod() throws Exception {
+
+        AlterTableAnalyzedStatement analysis =
+                (AlterTableAnalyzedStatement)analyze("alter table users set (\"gateway.local.sync\"=1)");
```
Comment (seut) [inline sql/src/test/java/io/crate/analyze/CreateAlterTableStatementAnalyzerTest.java]: would prefer '1s' here

File: docs/configuration.txt [hunk 7]
```diff
@@ -91,18 +91,33 @@ Blocks
 Translog
 --------
 
+.. note:
+
+  The translog provides a persistent log of all operations that have not been transferred (flushed) to disk.
+  Whenever a record is inserted into a table (or updated) that change is appended both to an in-memory buffer **and** the translog.
+  When refreshing a table the changes from the in-memory buffer are written to a new commit point (which is then searchable)
```
Comment (seut) [inline docs/configuration.txt]: pls remove refresh related things here

**Comment by seut** [comment]
otherwise ack if travis failures are unrelated

**Pull request merged**